punishments.js

```diff
@@ -389,7 +416,7 @@ setImmediate(() => {
  * @param {User} user
  * @param {Punishment} punishment
  * @param {?Set<string>} recursionKeys
```

Comment:
PS current system is to put the `?` at the end: `Set<string>?`

Mostly, this makes arrays less ambiguous, since `string?[]` and `string[]?` are different things.